java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/JdbcToArrowTestHelper.java

```diff
@@ -166,6 +173,41 @@ public static void assertNullValues(BaseValueVector vector, int rowCount) {
     }
   }
 
+  public static void assertFieldMetadataIsEmpty(VectorSchemaRoot schema) {
+    assertNotNull(schema);
+    assertNotNull(schema.getSchema());
+    assertNotNull(schema.getSchema().getFields());
+
+    for (Field field : schema.getSchema().getFields()) {
+      assertNotNull(field.getMetadata());
+      assertEquals(0, field.getMetadata().size());
+    }
+  }
+
+  public static void assertFieldMetadataMatchesResultSetMetadata(ResultSetMetaData rsmd, Schema schema)
+      throws SQLException {
+    assertNotNull(schema);
+    assertNotNull(schema.getFields());
+    assertNotNull(rsmd);
+
+    List<Field> fields = schema.getFields();
+
+    assertEquals(rsmd.getColumnCount(), fields.size());
+
+    // Vector columns are created in the same order as ResultSet columns.
+    for (int i = 0; i < rsmd.getColumnCount(); ++i) {
+      Map<String, String> metadata = fields.get(i).getMetadata();
+
+      assertNotNull(metadata);
+      assertEquals(4, metadata.size());
+
+      assertEquals(rsmd.getCatalogName(i + 1), metadata.get(Constants.SQL_CATALOG_NAME_KEY));
```

Comment:
might be cleaner if you start the loop at 1 instead of doing the addition in 4 places.

Reply:
Yup, I agree.  I had to use `i - 1` when getting the field metadata from the `VectorSchemaRoot`, but I agree that's 3 fewer lines of code with an offset.  Updated.